internal/agentdeployer/deployed_agent.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package agentdeployer
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+var ErrNotSupported error = errors.New("not supported")
+
+// DeployedAgent defines the interface for interacting with a service that has been deployed.
+type DeployedAgent interface {
+	// TearDown implements the logic for tearing down a service.
+	TearDown(ctx context.Context) error
+
+	// Signal sends a signal to the service.
+	Signal(ctx context.Context, signal string) error
+
+	// Info returns the current context from the service.
+	Info() AgentInfo
+
+	// SetInfo sets the current context for the service.
+	SetInfo(str AgentInfo) error
```

Comment:
On what cases this would return an error?

Reply:
I cannot think of any scenario where this could return an error.
I'll update this method to return nothing and update the corresponding implementations of `DeployedAgent`.

This was based on this interface (`DeployedService`):

https://github.com/elastic/elastic-package/blob/63624b4f4ccf07391099ef5c37b45fa4e331c42e/internal/servicedeployer/deployed_service.go#L26

Should we update that other interface too ?